extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java

```diff
@@ -428,8 +428,12 @@ private void submitTasks(
             runningTasks.put(entry.getKey(), task);
           }
         }
-        catch (EntryExistsException e) {
-          log.error("task %s already exsits", task);
+        catch (DruidException e) {
+          if ("invalidInput".equals(e.getErrorCode())) {
```

Comment:
This doesn't seem too neat. I wonder if it would be better to just have a `TaskExistsException` which extends `DruidException`. We could also double check this code to ensure that getting duplicates is not a common mode of operation. Then, we could simply not catch the `DruidException` altogether.

Reply:
Also, special handling needs to be done only for "duplicateEntry", not all kinds of "invalidInput". So I guess it might be good to just add a new `Failure` type for "duplicateEntry".